preop.cpp

```diff
@@ -36,16 +36,15 @@ char *handle_preop (char *ptr) {
 
 	//first get the name
 	name_end = ptr;
-	while (isalpha (*name_end))
+	while (!isspace(*name_end))

```

Comment:
If you go from a positive condition to a negative one, you need to add a check for the end of the buffer.